src/chainstate/burn/db/sortdb.rs

```diff
@@ -2133,20 +2148,84 @@ impl SortitionDB {
             PoxConstants::test_default(),
             true,
         )
     }
 
+    #[cfg(test)]
+    pub fn connect_v1(
+        path: &str,
+        first_block_height: u64,
+        first_burn_hash: &BurnchainHeaderHash,
+        first_burn_header_timestamp: u64,
+        readwrite: bool,
+    ) -> Result<SortitionDB, db_error> {
+        let create_flag = match fs::metadata(path) {
+            Err(e) => {
+                if e.kind() == ErrorKind::NotFound {
+                    // need to create
+                    if readwrite {
+                        true
+                    } else {
+                        return Err(db_error::NoDBError);
+                    }
+                } else {
+                    return Err(db_error::IOError(e));
+                }
+            }
+            Ok(_md) => false,
+        };
+
+        let index_path = db_mkdirs(path)?;
+        debug!(
+            "Connect/Open {} sortdb '{}' as '{}'",
+            if create_flag { "(create)" } else { "" },
+            index_path,
+            if readwrite { "readwrite" } else { "readonly" }
+        );
+
+        let marf = SortitionDB::open_index(&index_path)?;
+
+        let mut db = SortitionDB {
+            marf,
+            readwrite,
+            first_block_height,
+            first_burn_header_hash: first_burn_hash.clone(),
+            pox_constants: PoxConstants::test_default(),
+        };
+
+        if create_flag {
+            // instantiate!
+            db.instantiate_v1(
+                first_block_height,
+                first_burn_hash,
+                first_burn_header_timestamp,
+            )?;
+        } else {
+            // validate -- must contain the given first block and first block hash
+            let snapshot = SortitionDB::get_first_block_snapshot(db.conn())?;
+            if !snapshot.is_initial()
+                || snapshot.block_height != first_block_height
+                || snapshot.burn_header_hash != *first_burn_hash
+            {
+                error!("Invalid genesis snapshot: sn.is_initial = {}, sn.block_height = {}, sn.burn_hash = {}, expect.block_height = {}, expect.burn_hash = {}",
+                       snapshot.is_initial(), snapshot.block_height, &snapshot.burn_header_hash, first_block_height, first_burn_hash);
+                return Err(db_error::Corruption);
+            }
+        }
+
+        Ok(db)
+    }
+
     /// Validate all Stacks Epochs. Since this is data that always comes from a static variable,
     /// any invalid StacksEpoch structuring should result in a runtime panic.
     fn validate_epochs(epochs_ref: &[StacksEpoch]) -> Vec<StacksEpoch> {
         // sanity check -- epochs must all be contiguous, each epoch must be unique,
         // and the range of epochs should span the whole non-negative i64 space.
         let mut epochs = epochs_ref.to_vec();
         let mut seen_epochs = HashSet::new();
-        epochs.sort();
```

Comment:
Why is ts removed?

Reply:
Something was going on where it wasn't sorting by epoch_id.

Why would it not be that the epochs are passed in in the correct order?

I could update the code, but we can also just leave it as it is and require them to be passed in in order.

Either way the method contract can be updated to reflect the input requirements.

Reply:
> Something was going on where it wasn't sorting by epoch_id.

That sounds like a bug in this PR, then.  It was working before, so you should find and fix the bug instead of deleting non-regressed code.

> Why would it not be that the epochs are passed in in the correct order?

Why should we trust that no one will ever screw up their end of the API contract, when we can easily defend against it?

Reply:
I think the different sort behavior is an artifact of there being new fields in this struct in the 2.05 branch.

I can take a deeper look if we are curious.

Reply:
Right, this sounds indicative of something deeper.  Epochs should be totally ordered, and if something changed that removes this guarantee, then we have a problem.

Reply:
reverting the change. I forget now which test it was that is broken so we'll see.

Reply:
I reverted this change and no tests broke so I'm not sure what happened.

The test are all passing.